modules/core/src/test/scala/org/tesselation/trust/DataGenerator.scala

```diff
@@ -0,0 +1,68 @@
+package org.tesselation.trust
+
+import cats.effect.{Async, Ref}
+import cats.syntax.all._
+
+import scala.util.Random
+
+import org.tesselation.infrastructure.trust.{TrustEdge, TrustNode}
+
+class DataGenerator[F[_]: Async] {
+  private final val randomEffect: Ref[F, Random] = Ref.unsafe(new Random())
```

Comment:
Why is `Ref` needed here? I don't see you're updating it (so just val/def with pure/delayed random would be enough). Anyway, there is a `SecureRandom` instance from cats if you want an effect.

There is also a second option, to use a State monad for Random, but I think that's not the case here.

Reply:
I didn't write this Ref originally -- this was here before. I honestly don't think this code needs to be async either, so if you're okay with removing this that's fine with me. It's only supposed to be used in the tests for generating data before passing it somewhere else, so it really doesn't need to be captured.